pkg/core/resources/model/rest/resource.go

```diff
@@ -29,28 +29,26 @@ var _ json.Marshaler = &Resource{}
 var _ json.Unmarshaler = &Resource{}
 
 func (r *Resource) MarshalJSON() ([]byte, error) {
-	meta, err := json.Marshal(&r.Meta)
-	if err != nil {
-		return nil, err
-	}
-	if r.Spec == nil {
-		return meta, nil
+	var specBytes []byte
+	if r.Spec != nil {
+		var buf bytes.Buffer
+		if err := (&jsonpb.Marshaler{}).Marshal(&buf, r.Spec); err != nil {
+			return nil, err
+		}
+		specBytes = buf.Bytes()
 	}
 
-	var buf bytes.Buffer
-	if err := (&jsonpb.Marshaler{}).Marshal(&buf, r.Spec); err != nil {
+	metaJSON, err := json.Marshal(r.Meta)
+	if err != nil {
 		return nil, err
 	}
-	spec := buf.Bytes()
 
-	var obj map[string]json.RawMessage
-	if err := json.Unmarshal(meta, &obj); err != nil {
-		return nil, err
-	}
-	if err := json.Unmarshal(spec, &obj); err != nil {
-		return nil, err
+	if len(specBytes) == 0 || reflect.DeepEqual(specBytes, []byte{'{', '}'}) { // spec is nil or empty
```

Comment:
Let's replace
```
reflect.DeepEqual(specBytes, []byte{'{', '}'})
```
with
```
string(specBytes) == "{}"
```

`string(specBytes)` is always safe since strings in Golang hold arbitrary bytes rather than Unicode, UTF-8, etc.